### PR TITLE
Add alert response

### PR DESCRIPTION
**Pull request by sebastianmueller01**

### CODE REVIEW

File: Core/Inc/packData.h [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #ifndef INC_PACKDATA_H_
 #define INC_PACKDATA_H_
 
-#define MAX_CELL_IMBALANCE_V    0.1f
+#define MAX_CELL_IMBALANCE_V    0.2f
```
Comment (sebastianmueller01) [inline Core/Inc/packData.h]: This should probably be moved directly to `alerts.h`

File: Core/Inc/shared.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#ifndef INC_SHARED_H_
+#define INC_SHARED_H_
+
+// Responses for a specific alert
+typedef enum
+{
+    INFO_ONLY = 0,
```
Comment (sebastianmueller01) [inline Core/Inc/shared.h]: Add more comments

File: Core/Src/alerts.c [hunk 3]
```diff
@@ -95,7 +171,7 @@ void runAlertMonitor(Bms_S* bms, Alert_S* alert)
         }
 
         // Determine if alert was set
-        if (checkTimerExpired(&alert->alertTimer))
+        if (checkTimerExpired(&alert->alertTimer) && (!(alert->alertTimer.timThreshold <= 0) || alert->alertConditionPresent(bms)))
```
Comment (sebastianmueller01) [inline Core/Src/alerts.c:174]: Add a comment explaining this

File: Core/Src/bmb.c [hunk 4]
```diff
@@ -165,12 +242,16 @@ void updateBmbData(Bmb_S* bmb, uint32_t numBmbs)
...
-				uint16_t scanCtrlData = (recvBuffer[4 + 2*j] << 8) | recvBuffer[3 + 2*j];
+				// Extract register contents from receive buffer
+				uint16_t scanCtrlData = getValueFromBuffer(recvBuffer, j);
 				allBmbScanDone &= ((scanCtrlData & 0xA000) == 0xA000);	// Verify SCANDONE and DATARDY bits
+				// TODO: We may want to still read in data if not all scans were completed. For example if only one BMB
+				// failed to complete the scan in time we still have 6 BMBs with good data. 
 			}
 			if (!allBmbScanDone)
 			{
 				// Do we want to try to restart the scans here?
```
Comment (sebastianmueller01) [inline Core/Src/bmb.c]: Update comment explaining why we are restarting scans here

File: Core/Src/bmb.c [hunk 5]
```diff
@@ -182,52 +263,61 @@ void updateBmbData(Bmb_S* bmb, uint32_t numBmbs)
...
+					bmb[bmbIdx].brickV[i] = brickV;
+					bmb[bmbIdx].brickVStatus[i] = is14BitSensorRailed(brickVRaw) ? BAD : GOOD;
 				}
 			}
 			else
 			{
 				Debug("Error during cellReg readAll!\n");
 
-				// Failed to acquire data. Set status to MIA
+				// Failed to acquire data. Set status to  BAD
```
Comment (sebastianmueller01) [inline Core/Src/bmb.c]: Get rid of extra space

File: Core/Src/bmb.c [hunk 6]
```diff
@@ -182,52 +263,61 @@ void updateBmbData(Bmb_S* bmb, uint32_t numBmbs)
...
 			Debug("Error during VBLOCK readAll!\n");
 
-			// Failed to acquire data. Set status to MIA
+			// Failed to acquire data. Set status to BAD
 			for (int32_t j = 0; j < numBmbs; j++)
 			{
-				bmb[j].stackVStatus = MIA;
+				// Convert from frame index (starts with last BMB) to bmb index (starts with first BMB) 
+				const uint32_t bmbIdx = numBmbs - j - 1;
+				bmb[bmbIdx].segmentVStatus =BAD;
```
Comment (sebastianmueller01) [inline Core/Src/bmb.c]: Add space before BAD

File: Core/Src/bmb.c [hunk 7]
```diff
@@ -513,7 +533,7 @@ int32_t detectBmbDaisyChainBreak(Bmb_S* bmb, uint32_t numBmbs)
 		for (uint32_t i = 0; i < bmbIdx + 1; i++)
 		{
 			// Read model number in [15:4]
-			uint32_t versionRegister = ((recvBuffer[4 + 2*i] << 8) | recvBuffer[3 + 2*i]) >> 4;
+			uint32_t versionRegister = getValueFromBuffer(recvBuffer, i) >> 4;
 			if (versionRegister != 0x843)
```
Comment (sebastianmueller01) [inline Core/Src/bmb.c]: Replace 0x843 with a #define

File: Core/Src/mainTask.c [hunk 8]
```diff
@@ -37,7 +37,7 @@ uint32_t numBmbs = 0;
 static uint32_t initRetries = 5;
 static uint32_t lastUpdateMain = 0;
 
-bool balancingEnabled = false;
```
Comment (sebastianmueller01) [inline Core/Src/mainTask.c]: Set back to false probably

File: Core/Inc/alerts.h [hunk 9]
```diff
@@ -5,21 +5,29 @@
...
 
-#define CELL_IMBALANCE_ALERT_SET_TIME_MS    1000
-#define CELL_IMBALANCE_ALERT_CLEAR_TIME_MS  1000
+#define UNDERVOLTAGE_WARNING_ALERT_SET_TIME_MS    2000
+#define UNDERVOLTAGE_WARNING_ALERT_CLEAR_TIME_MS  2000
+
+#define UNDERVOLTAGE_FAULT_ALERT_SET_TIME_MS      2000
+#define UNDERVOLTAGE_FAULT_ALERT_CLEAR_TIME_MS    2000
+
+#define MAX_CELL_IMBALANCE_V                      0.2f
```
Comment (isaiahwild42) [inline Core/Inc/alerts.h:28]: Does this belong in cellData.h?

Comment (sebastianmueller01) [reply Core/Inc/alerts.h:28]: No, since it isn't inherently information about the cell, just the point where we consider something to have gone wrong

**Pull request merged**